_studio/shared/src/libmfx_allocator_vaapi.cpp

```diff
@@ -110,6 +110,100 @@ unsigned int ConvertMfxFourccToVAFormat(mfxU32 fourcc)
     }
 }
 
+static mfxStatus ReallocImpl(mfxDefaultAllocatorVAAPI::mfxWideHWFrameAllocator* pSelf, vaapiMemIdInt *vaapi_mid, mfxFrameSurface1 *surf)
+{
+    VAStatus  va_res  = VA_STATUS_SUCCESS;
+    unsigned int va_fourcc = 0;
+    mfxU32 fourcc = surf->Info.FourCC;
+
+    // VP8 hybrid driver has weird requirements for allocation of surfaces/buffers for VP8 encoding
+    // to comply with them additional logic is required to support regular and VP8 hybrid allocation pathes
+    mfxU32 mfx_fourcc = ConvertVP8FourccToMfxFourcc(fourcc);
+    va_fourcc = ConvertMfxFourccToVAFormat(mfx_fourcc);
+    if (!va_fourcc || ((VA_FOURCC_NV12 != va_fourcc) &&
+                       (VA_FOURCC_YV12 != va_fourcc) &&
+                       (VA_FOURCC_YUY2 != va_fourcc) &&
+                       (VA_FOURCC_ARGB != va_fourcc) &&
+                       (VA_FOURCC_ABGR != va_fourcc) &&
+                       (VA_FOURCC_P208 != va_fourcc) &&
+                       (VA_FOURCC_P010 != va_fourcc) &&
+                       (VA_FOURCC_YUY2 != va_fourcc) &&
+#if (MFX_VERSION >= 1027)
+                       (VA_FOURCC_Y210 != va_fourcc) &&
+                       (VA_FOURCC_Y410 != va_fourcc) &&
+#endif
+                       (VA_FOURCC_AYUV != va_fourcc) ))
+    {
+        return MFX_ERR_UNSUPPORTED;
+    }
+
+    VASurfaceID surfaces[1];
+    VASurfaceAttrib attrib[2];
+
+    surfaces[0] = *vaapi_mid->m_surface;
+    if (MFX_FOURCC_P8 == vaapi_mid->m_fourcc)
+    {
+        vaDestroyBuffer(pSelf->pVADisplay, surfaces[0]);
+    }
+    vaDestroySurfaces(pSelf->pVADisplay, &surfaces[0], 1);
+
+    *vaapi_mid->m_surface = 0;
+
+    unsigned int format;
+    int attrCnt = 0;
```

Comment:
misleading name. it is not count, it is index. it leads to error on line 177. I suggest to use ```std::vector``` instead, see my variant of this function

Reply:
I like simple array version. vector is not needed here.